### ISSUE TITLE
Offline Translation with Qt ts files doesn't work 

### ISSUE THREAD

**Issue by ChrSchwaer** [translate/pootle#5418]
Using:
Pootle 2.7.6
Translate Toolkit 1.13.0.

We use a Pootle project with Qt .ts Files as Source.

If we want to download a language file for offline translation we can't open them with the Qt tool QtLinguist! We good a error on open:
Thats the Error Message
`Unexpected tag <message> at /Users/christian_schw/Downloads/ro_RO.ts:4:11` 

The Downloaded file looks like this:

```
<?xml version='1.0' encoding='utf-8'?>
<!DOCTYPE TS>
<TS language="ro">
  <message><source>Infos</source><translation></translation>
    </message>
  <message><source>Version:</source><translation></translation>
    </message>
...
</TS>
```

How can i fix them?

**Comment by dwaynebailey** [comment]
@ChrSchwaer could you try the following things and also give us some of the following information:

* What version of QtLinguist are you running
* Could you try update Translate Toolkit to 2.0.0b6
* Could you try remove the first empty `<translation></translation>` and confirm if the reported error moves (in which case removing all empty `<translation>` tags would make the file editable.

**Comment by ChrSchwaer** [comment]
- QtLinguist Version 5.5.1
- Remove the <translation></translation> doesn't matter. Got the same message!

- I didn't update the Translate Toolkit! It's still a Beta Version! Should i do that?
   When is the Pootle release V2.8.0 planed?

**Comment by dwaynebailey** [comment]
@ChrSchwaer thanks for the version.  Now at least we know what [.ts format spec](http://doc.qt.io/qt-5/linguist-ts-file-format.html>) weshould be trying to follow

Can you actually paste the error message? What I was after was not to see if that loaded the file ut to see if reported failure line number changed.  If it changed we'd know that its the source of the error and removing all of those empty <translations> would fix it.  I'm afraid I don't have linguist to test myself so I need you to be quite specific about tests I'm asking you to run and to report the actual messages you receive.

Translate Toolkit is beta but it includes a number of .ts changes so it would be a good method to determine if those eliminate the issue.  The format handling is provided by Translate Toolkit and not Pootle.

We're pretty close to releasing the first v2.8.0 release candidate and we run it live on a number of sites.

**Comment by ChrSchwaer** [comment]
![linguist_error](https://cloud.githubusercontent.com/assets/22295392/19600509/9f7b1bfe-97a5-11e6-987d-b6c36ea3c27e.png)

The Line number didn't change after delete the <translation></translation> tags!

The correct .ts Files should look like this:
```
<?xml version='1.0' encoding='utf-8'?>
<!DOCTYPE TS>
<TS version="2.1" language="ro">
<context>
    <name></name>
    <message id="ui.infosTitle">
        <source>Infos</source>
        <translation type="unfinished"></translation>
    </message>
    <message id="ui.appVersion">
        <source>Version:</source>
        <translation type="unfinished"></translation>
    </message>
...
</context>
</TS>
```
Downloading the .ts Files from translation memory works as well. Just the offline export doesn't work!

**Comment by ChrSchwaer** [comment]
Now we updated the Translate Toolkit but we still got the same error!
Pootle 2.7.6 uses Translate Toolkit 2.0.0b6. 

```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE TS>
<TS language="de">
  <message><source>Ok</source><translation>OK</translation>
    </message>
  <message><source>Cancel</source><translation>Abbrechen</translation>
    </message>
</TS>

**Comment by ChrSchwaer** [comment]
Are there any news about this?

**Comment by dwaynebailey** [comment]
@ChrSchwaer I don't think we have enough info to understand what actually causes the .ts file to fail. Since this is on an export for offlines, which we're aware does not work correctly for non-PO files there is much more that would need to be fixed to meet your needs.  At the moment we're quite pressed getting 2.8.0 stable and out the door that we aren't able prioritise this at the moment.

Or course you could change that by submitting a PR that addresses the problem or engaging us commercially to fix the problem.

**Comment by khagaroth** [comment]
> I don't think we have enough info to understand what actually causes the .ts file to fail.

Wrong XML structure. The `<message>` elements must be inside a `<context>` element. Also, the first element of `<context>` must be a `<name>`. QT does have a DTD for the format, you should check it.

**Comment by ChrSchwaer** [comment]
Qt DTD Description: [http://doc.qt.io/qt-5/linguist-ts-file-format.html](url)
